src/modules/rlm_redis_ippool/rlm_redis_ippool.c

```diff
@@ -211,14 +211,23 @@ static char lua_alloc_cmd[] =
 	 */
 	"exists = redis.call('GET', owner_key);" EOL							/* 8 */
 	"if exists then" EOL										/* 9 */
-	"  local expires_in = tonumber(redis.call('ZSCORE', pool_key, exists) - ARGV[1])" EOL		/* 10 */
-	"  if expires_in > 0 then" EOL									/* 11 */
-	"    ip = redis.call('HMGET', '{' .. KEYS[1] .. '}:"IPPOOL_ADDRESS_KEY":' .. exists, 'device', 'range', 'counter')" EOL	/* 12 */
+	"  local expires = tonumber(redis.call('ZSCORE', pool_key, exists))" EOL			/* 10 */
+	"  local static = expires >= 2^" STRINGIFY(IPPOOL_STATIC_BIT) EOL
+	"  local expires_in = expires - (static and 2^" STRINGIFY(IPPOOL_STATIC_BIT) " or 0) - ARGV[1]" EOL
+	"  if expires_in > 0 or static then" EOL							/* 11 */
+	"    ip = redis.call('HMGET', '{' .. KEYS[1] .. '}:"IPPOOL_ADDRESS_KEY":' .. exists, 'device', 'range', 'counter', 'gateway')" EOL	/* 12 */
 	"    if ip and (ip[1] == ARGV[3]) then" EOL							/* 13 */
 	"      if expires_in < tonumber(ARGV[2]) then" EOL						/* 14 */
-	"        redis.call('ZADD', pool_key, 'XX', ARGV[1] + ARGV[2], exists)" EOL			/* 15 */
+	"        redis.call('ZADD', pool_key, 'XX', ARGV[1] + ARGV[2] + (static and 2^" STRINGIFY(IPPOOL_STATIC_BIT) " or 0), exists)" EOL	/* 15 */
 	"        expires_in = tonumber(ARGV[2])" EOL							/* 16 */
 	"      end" EOL											/* 17 */
+
+	/*
+	 *	Ensure gateway is set correctly
+	 */
+	"      if ARGV[4] ~= ip[4] then" EOL
```

Comment:
Ah yes, good catch.  I guess in the majority of cases the gateway doesn't change, but it might if you were moving around between APs or something

Reply:
And for static assignments - they never hit the gateway assignment at the end of the script